src/Perlang.Interpreter/PerlangInterpreter.cs

```diff
@@ -636,23 +685,30 @@ public object VisitUnaryPostfixExpr(Expr.UnaryPostfix expr)
             }
 
             // The nullability check has been taken care of by IsValidNumberType() for us.
-            dynamic previousValue = left!;
             var variable = (Expr.Identifier)expr.Left;
-            object value;
 
-            switch (expr.Operator.Type)
+            object value = left switch
             {
-                case PLUS_PLUS:
-                    value = previousValue + 1;
-                    break;
-
-                case MINUS_MINUS:
-                    value = previousValue - 1;
-                    break;
-
-                default:
-                    throw new RuntimeError(expr.Operator, $"Unsupported operator encountered: {expr.Operator.Type}");
-            }
+                int previousIntValue => expr.Operator.Type switch
+                {
+                    PLUS_PLUS => previousIntValue + 1,
+                    MINUS_MINUS => previousIntValue - 1,
+                    _ => throw new RuntimeError(expr.Operator, $"Unsupported operator encountered: {expr.Operator.Type}")
+                },
+                long previousLongValue => expr.Operator.Type switch
+                {
+                    PLUS_PLUS => previousLongValue + 1,
+                    MINUS_MINUS => previousLongValue - 1,
+                    _ => throw new RuntimeError(expr.Operator, $"Unsupported operator encountered: {expr.Operator.Type}")
+                },
+                double previousDoubleValue => expr.Operator.Type switch
+                {
+                    PLUS_PLUS => previousDoubleValue + 1,
+                    MINUS_MINUS => previousDoubleValue - 1,
+                    _ => throw new RuntimeError(expr.Operator, $"Unsupported operator encountered: {expr.Operator.Type}")
+                },
+                _ => throw new RuntimeError(expr.Operator, $"Internal runtime: Unsupported type {ToTypeKeywordNew(left?.GetType())} encountered with {expr.Operator.Type} operator")
```

Comment:
Should also use https://github.com/perlang-org/perlang/pull/227/files#r750651560